packages/node_modules/@node-red/editor-client/src/js/history.js

```diff
@@ -453,10 +453,48 @@ RED.history = (function() {
                                     RED.events.emit("nodes:change",newConfigNode);
                                 }
                             });
+                        } else if (i === "env" && ev.node.type.indexOf("subflow:") === 0) {
+                            // Subflow can have config node in node.env
+                            let nodeList = ev.node.env || [];
+                            nodeList = nodeList.reduce((list, prop) => {
+                                if (prop.type === "conf-type" && prop.value) {
+                                    list.push(prop.value);
+                                }
+                                return list;
+                            }, []);
+
+                            nodeList.forEach(function(id) {
+                                const configNode = RED.nodes.node(id);
+                                if (configNode) {
+                                    if (configNode.users.indexOf(ev.node) !== -1) {
+                                        configNode.users.splice(configNode.users.indexOf(ev.node), 1);
+                                        RED.events.emit("nodes:change", configNode);
+                                    }
+                                }
+                            });
+
+                            nodeList = ev.changes.env || [];
+                            nodeList = nodeList.reduce((list, prop) => {
+                                if (prop.type === "conf-type" && prop.value) {
+                                    list.push(prop.value);
+                                }
+                                return list;
+                            }, []);
+
+                            nodeList.forEach(function(id) {
+                                const configNode = RED.nodes.node(id);
+                                if (configNode) {
+                                    if (configNode.users.indexOf(ev.node) === -1) {
+                                        configNode.users.push(ev.node);
+                                        RED.events.emit("nodes:change", configNode);
+                                    }
+                                }
+                            });
                         }
```

Comment:
GH didn't want this one
```suggestion
                        }

                        if (i === "credentials" && ev.changes[i]) {
                            for (const [key, value] of Object.entries(ev.changes[i])) {
                                ev.node.credentials[key] = value;
                            }
                        } else {
                            ev.node[i] = ev.changes[i];
                        }
```

Reply:
I don't know how to interpret the comment ''GH didn't want this one'' so I'm not sure what this suggested change is for. Can you explain?

Reply:
Okay. What is the purpose of the change? What is it fixing?

Reply:
I see now this is trying to fix the undo of credential changes. But I think this suggested change hasn't spanned the lines it's intended to replace - if I commit it as-is, it will be followed by `ev.node[i] = ev.changes[i];` again.

I'll propose the same change, but target the line to replace.